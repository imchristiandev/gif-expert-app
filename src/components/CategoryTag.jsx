import React from 'react'

export const CategoryTag = ({name, onDeleteCategory}) => {

  const handleDeleteCategory = () => {
    onDeleteCategory(name);
  }
  
  return (
    <span>
        {name}
        <button onClick={handleDeleteCategory}>âœ•</button>
    </span>
  )
}
