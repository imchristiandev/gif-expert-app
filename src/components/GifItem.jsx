import { LazyLoadImage } from 'react-lazy-load-image-component'
import PropTypes from 'prop-types'
import 'react-lazy-load-image-component/src/effects/blur.css';

export const GifItem = ( { title, url } ) => {
  return (
    <div className='card'>
      <LazyLoadImage
        alt={title}
        src={url} // use normal <img> attributes as props
        effect="blur"
      />
        <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
