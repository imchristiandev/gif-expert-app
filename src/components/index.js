export * from './AddCategory'
export * from './GifGrid'
export * from './GifItem'
export * from './CategoryTagList'
export * from './CategoryTag'