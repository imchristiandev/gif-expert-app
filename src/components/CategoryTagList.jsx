import { CategoryTag } from "./CategoryTag";

export const CategoryTagList = ({categories, onDeleteCategory}) => {

  const handleDeleteCategory = (category) => {
    onDeleteCategory(category)
    
  }
  return (
    <div className="tag">
      {
        categories.map((category) => 
          <CategoryTag 
            key={category} 
            name={category}
            onDeleteCategory={handleDeleteCategory}
          />
        )
      }
    </div>
  )
}
