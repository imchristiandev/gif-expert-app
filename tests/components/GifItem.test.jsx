import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('Pruebas en <GifItemApp />', () => { 

    const title="Saitama";
    const url="https://media3.giphy.com/media/QAU5Bd0Js9w3arZ4UD/giphy.gif?cid=1d513e8bwzcndfxg5fqct1awzft9zw5rp0ghydyobrdyfj5s&rid=giphy.gif&ct=g"

    test('Debe de hacer match con el snapshot', () => { 
        const { container } = render( <GifItem title={title} url={url}/> )
        expect( container ).toMatchSnapshot()
    })

    test('Debe de mostrar con el url y el alt indicado', () => { 
        render( <GifItem title={title} url={url}/> )
        //screen.debug()
        const { src, alt } = screen.getByRole('img')
        expect( src ).toBe( url )
        expect( alt ).toBe( title )
    })

    test('Debe de mostrar el tÃ­tulo en el componente', () => { 
        render( <GifItem title={ title } url={ url }/> )
        expect( screen.getByText( title ) ).toBeTruthy()   
    })
})