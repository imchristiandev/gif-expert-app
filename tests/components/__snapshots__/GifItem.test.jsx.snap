// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItemApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <span
      class=" lazy-load-image-background blur"
      style="color: transparent; display: inline-block;"
    >
      <img
        alt="Saitama"
        src="https://media3.giphy.com/media/QAU5Bd0Js9w3arZ4UD/giphy.gif?cid=1d513e8bwzcndfxg5fqct1awzft9zw5rp0ghydyobrdyfj5s&rid=giphy.gif&ct=g"
      />
    </span>
    <p>
      Saitama
    </p>
  </div>
</div>
`;
